Problem with CDC and Postgres [HZ-2086]

Hi,

I have a problem with CDC and a Postgres DB.

I get a error when running the job:
```
2023-02-13 13:54:54 2023-02-13 12:54:54,470 [ INFO] [hz.vibrant_curran.cached.thread-2] [c.h.j.i.MasterJobContext]: [192.168.0.2]:5701 [hello-world] [5.2.1] Start executing job 'postgres-monitor-company_profile', execution 0966-f4c7-c140-0001, execution graph in DOT format:
2023-02-13 13:54:54 digraph DAG {
2023-02-13 13:54:54     "source" [localParallelism=1];
2023-02-13 13:54:54     "loggerSink" [localParallelism=1];
2023-02-13 13:54:54     "source" -> "loggerSink" [queueSize=1024];
2023-02-13 13:54:54 }
2023-02-13 13:54:54 HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
2023-02-13 13:54:54 2023-02-13 12:54:54,535 [ INFO] [hz.vibrant_curran.cached.thread-2] [c.h.j.i.JobExecutionService]: [192.168.0.2]:5701 [hello-world] [5.2.1] Execution plan for jobId=0966-f4c7-8c40-0001, jobName='postgres-monitor-company_profile', executionId=0966-f4c7-c140-0001 initialized
2023-02-13 13:54:54 2023-02-13 12:54:54,540 [ INFO] [hz.vibrant_curran.cached.thread-2] [c.h.j.i.JobExecutionService]: [192.168.0.2]:5701 [hello-world] [5.2.1] Start execution of job 'postgres-monitor-company_profile', execution 0966-f4c7-c140-0001 from coordinator [192.168.0.2]:5701
2023-02-13 13:54:54 2023-02-13 12:54:54,559 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [c.h.j.c.i.ChangeRecordCdcSourceP]: [192.168.0.2]:5701 [hello-world] [5.2.1] [postgres-monitor-company_profile/source#0] source has 'retry strategy' set to 'max attempts = 0, interval function = IntervalFunction (constant of 5000ms)'
2023-02-13 13:54:54 2023-02-13 12:54:54,560 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [c.h.j.c.i.ChangeRecordCdcSourceP]: [192.168.0.2]:5701 [hello-world] [5.2.1] [postgres-monitor-company_profile/source#0] source has 'commit period' set to '10000000000'
2023-02-13 13:54:54 2023-02-13 12:54:54,560 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [c.h.j.c.i.ChangeRecordCdcSourceP]: [192.168.0.2]:5701 [hello-world] [5.2.1] [postgres-monitor-company_profile/source#0] source has 'clear state on reconnect' set to 'false'
2023-02-13 13:54:54 2023-02-13 12:54:54,573 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [c.h.j.c.i.ChangeRecordCdcSourceP]: [192.168.0.2]:5701 [hello-world] [5.2.1] [postgres-monitor-company_profile/source#0] starting new task with config: {connector.class=io.debezium.connector.postgresql.PostgresConnector, database.dbname=postgres, database.user=postgres, db.specific.extra.fields=schema, database.server.name=5664d41e-18a6-437f-93d4-d10eb54d2e6c, sequence.extractor.class=com.hazelcast.jet.cdc.postgres.impl.PostgresSequenceExtractor, plugin.name=pgoutput, database.port=30000, table.whitelist=public.company_profile, tombstones.on.delete=false, database.hostname=newworkjob.de, database.password=7y8q0olesb, name=source, database.history=com.hazelcast.jet.cdc.impl.CdcSourceP$DatabaseHistoryImpl, snapshot.mode=initial}
2023-02-13 13:54:54 2023-02-13 12:54:54,626 [ WARN] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.PostgresConnectorConfig]: Configuration property 'truncate.handling.mode' is deprecated and will be removed soon. If you wish to retain skipped truncate functionality, please configure 'skipped.operations' with "t".
2023-02-13 13:54:54 2023-02-13 12:54:54,626 [ WARN] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.PostgresConnectorConfig]: Configuration property 'truncate.handling.mode' is deprecated and will be removed in future versions. Please use 'skipped.operations' instead.
2023-02-13 13:54:54 2023-02-13 12:54:54,632 [ WARN] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.Configuration]: Using configuration property "table.whitelist" is deprecated and will be removed in future versions. Please use "table.include.list" instead.
2023-02-13 13:54:54 2023-02-13 12:54:54,633 [ WARN] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.PostgresConnectorConfig]: Configuration property 'toasted.value.placeholder' is deprecated and will be removed in future versions. Please use 'unavailable.value.placeholder' instead.
2023-02-13 13:54:54 2023-02-13 12:54:54,633 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]: Starting PostgresConnectorTask with configuration:
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    connector.class = io.debezium.connector.postgresql.PostgresConnector
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.dbname = postgres
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.user = postgres
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    db.specific.extra.fields = schema
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.server.name = 5664d41e-18a6-437f-93d4-d10eb54d2e6c
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    sequence.extractor.class = com.hazelcast.jet.cdc.postgres.impl.PostgresSequenceExtractor
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    plugin.name = pgoutput
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.port = 30000
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    table.whitelist = public.company_profile
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    tombstones.on.delete = false
2023-02-13 13:54:54 2023-02-13 12:54:54,634 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.hostname = newworkjob.de
2023-02-13 13:54:54 2023-02-13 12:54:54,635 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.password = ********
2023-02-13 13:54:54 2023-02-13 12:54:54,635 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    name = source
2023-02-13 13:54:54 2023-02-13 12:54:54,635 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    database.history = com.hazelcast.jet.cdc.impl.CdcSourceP$DatabaseHistoryImpl
2023-02-13 13:54:54 2023-02-13 12:54:54,635 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]:    snapshot.mode = initial
2023-02-13 13:54:54 2023-02-13 12:54:54,897 [ INFO] [pool-2-thread-1] [i.d.j.JdbcConnection]: Connection gracefully closed
2023-02-13 13:54:55 2023-02-13 12:54:55,418 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]: No previous offsets found
2023-02-13 13:54:55 2023-02-13 12:54:55,486 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.PostgresConnectorTask]: user 'postgres' connected to database 'postgres' on PostgreSQL 15.1 on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit with roles:
2023-02-13 13:54:55     role 'pg_read_all_settings' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_database_owner' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_stat_scan_tables' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_checkpoint' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_write_server_files' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_read_all_data' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_write_all_data' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_monitor' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_read_server_files' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_execute_server_program' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_read_all_stats' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'pg_signal_backend' [superuser: false, replication: false, inherit: true, create role: false, create db: false, can log in: false]
2023-02-13 13:54:55     role 'postgres' [superuser: true, replication: true, inherit: true, create role: true, create db: true, can log in: true]
2023-02-13 13:54:55 2023-02-13 12:54:55,522 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.c.PostgresConnection]: Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{0/A23A73B8}, catalogXmin=1060498]
2023-02-13 13:54:55 2023-02-13 12:54:55,523 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.PostgresConnectorTask]: No previous offset found
2023-02-13 13:54:55 2023-02-13 12:54:55,523 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.p.s.InitialSnapshotter]: Taking initial snapshot for new datasource
2023-02-13 13:54:55 2023-02-13 12:54:55,575 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.u.Threads]: Requested thread factory for connector PostgresConnector, id = 5664d41e-18a6-437f-93d4-d10eb54d2e6c named = change-event-source-coordinator
2023-02-13 13:54:55 2023-02-13 12:54:55,581 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.u.Threads]: Creating thread debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator
2023-02-13 13:54:55 2023-02-13 12:54:55,586 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Metrics registered
2023-02-13 13:54:55 2023-02-13 12:54:55,586 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Context created
2023-02-13 13:54:55 2023-02-13 12:54:55,590 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.s.InitialSnapshotter]: Taking initial snapshot for new datasource
2023-02-13 13:54:55 2023-02-13 12:54:55,590 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSnapshotChangeEventSource]: According to the connector configuration data will be snapshotted
2023-02-13 13:54:55 2023-02-13 12:54:55,591 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 1 - Preparing
2023-02-13 13:54:55 2023-02-13 12:54:55,859 [ WARN] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.SchemaNameAdjuster]: The Kafka Connect schema name '5664d41e-18a6-437f-93d4-d10eb54d2e6c.public.company_profile.Value' is not a valid Avro schema name, so replacing with '_664d41e_18a6_437f_93d4_d10eb54d2e6c.public.company_profile.Value'
2023-02-13 13:54:55 2023-02-13 12:54:55,859 [ WARN] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.SchemaNameAdjuster]: The Kafka Connect schema name '5664d41e-18a6-437f-93d4-d10eb54d2e6c.public.company_profile.Key' is not a valid Avro schema name, so replacing with '_664d41e_18a6_437f_93d4_d10eb54d2e6c.public.company_profile.Key'
2023-02-13 13:54:55 2023-02-13 12:54:55,868 [ WARN] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.SchemaNameAdjuster]: The Kafka Connect schema name '5664d41e-18a6-437f-93d4-d10eb54d2e6c.public.company_profile.Envelope' is not a valid Avro schema name, so replacing with '_664d41e_18a6_437f_93d4_d10eb54d2e6c.public.company_profile.Envelope'
2023-02-13 13:54:55 2023-02-13 12:54:55,871 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 2 - Determining captured tables
2023-02-13 13:54:55 2023-02-13 12:54:55,933 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_simple_triggers to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,933 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.filter to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,933 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.file to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,933 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.tag_editor to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_fired_triggers to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.match to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_triggers to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.company_profile to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_locks to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.user_profile to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.daemon to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_paused_trigger_grps to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.like to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.flyway_schema_history to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_simprop_triggers to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.user to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_job_details to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.subject_area_relation to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.job_profile to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.tasks to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,934 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.profile to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_blob_triggers to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_calendars to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.tag to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.tag_request to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.subject_area_skill to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_scheduler_state to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.contact to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.subject_area to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.theme to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.qrtz_cron_triggers to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,935 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Adding table public.distance to the list of capture schema tables
2023-02-13 13:54:55 2023-02-13 12:54:55,938 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 3 - Locking captured tables [public.company_profile]
2023-02-13 13:54:55 2023-02-13 12:54:55,938 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 4 - Determining snapshot offset
2023-02-13 13:54:55 2023-02-13 12:54:55,939 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSnapshotChangeEventSource]: Creating initial offset context
2023-02-13 13:54:56 2023-02-13 12:54:56,000 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSnapshotChangeEventSource]: Read xlogStart at 'LSN{0/A2E768D8}' from transaction '1065078'
2023-02-13 13:54:56 2023-02-13 12:54:56,148 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSnapshotChangeEventSource]: Read xlogStart at 'LSN{0/A2E768D8}' from transaction '1065078'
2023-02-13 13:54:56 2023-02-13 12:54:56,148 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 5 - Reading structure of captured tables
2023-02-13 13:54:56 2023-02-13 12:54:56,149 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSnapshotChangeEventSource]: Reading structure of schema 'public' of catalog 'postgres'
2023-02-13 13:54:56 2023-02-13 12:54:56,523 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 6 - Persisting schema history
2023-02-13 13:54:56 2023-02-13 12:54:56,523 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshot step 7 - Snapshotting data
2023-02-13 13:54:56 2023-02-13 12:54:56,524 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Snapshotting contents of 1 tables while still in transaction
2023-02-13 13:54:56 2023-02-13 12:54:56,524 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]: Exporting data from table 'public.company_profile' (1 of 1 tables)
2023-02-13 13:54:56 2023-02-13 12:54:56,525 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]:       For table 'public.company_profile' using select statement: 'SELECT "id", "createdAt", "updatedAt", "lastUpdatedBy", "userId", "profileType", "activated", "name", "logo", "size", "location", "branches", "homeOffice", "workTime", "agile", "hierarchie", "duCulture", "interNational", "euCitizen", "benefits" FROM "public"."company_profile"'
2023-02-13 13:54:56 2023-02-13 12:54:56,613 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.r.RelationalSnapshotChangeEventSource]:       Finished exporting 3 records for table 'public.company_profile'; total duration '00:00:00.089'
2023-02-13 13:54:56 2023-02-13 12:54:56,644 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.s.AbstractSnapshotChangeEventSource]: Snapshot - Final stage
2023-02-13 13:54:56 2023-02-13 12:54:56,645 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Snapshot ended with SnapshotResult [status=COMPLETED, offset=PostgresOffsetContext [sourceInfoSchema=Schema{io.debezium.connector.postgresql.Source:STRUCT}, sourceInfo=source_info[server='5664d41e-18a6-437f-93d4-d10eb54d2e6c'db='postgres', lsn=LSN{0/A2E768D8}, txId=1065078, timestamp=2023-02-13T12:54:56.613Z, snapshot=FALSE, schema=public, table=company_profile], lastSnapshotRecord=true, lastCompletelyProcessedLsn=null, lastCommitLsn=null, streamingStoppingLsn=null, transactionContext=TransactionContext [currentTransactionId=null, perTableEventCount={}, totalEventCount=0], incrementalSnapshotContext=IncrementalSnapshotContext [windowOpened=false, chunkEndPosition=null, dataCollectionsToSnapshot=[], lastEventKeySent=null, maximumKey=null]]]
2023-02-13 13:54:56 2023-02-13 12:54:56,649 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Connected metrics set to 'true'
2023-02-13 13:54:56 2023-02-13 12:54:56,875 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSchema]: REPLICA IDENTITY for 'public.company_profile' is 'DEFAULT'; UPDATE and DELETE events will contain previous values only for PK columns
2023-02-13 13:54:56 2023-02-13 12:54:56,877 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Starting streaming
2023-02-13 13:54:56 2023-02-13 12:54:56,877 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresStreamingChangeEventSource]: Retrieved latest position from stored offset 'LSN{0/A2E768D8}'
2023-02-13 13:54:56 2023-02-13 12:54:56,878 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.c.WalPositionLocator]: Looking for WAL restart position for last commit LSN 'null' and last change LSN 'LSN{0/A2E768D8}'
2023-02-13 13:54:56 2023-02-13 12:54:56,878 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.c.PostgresReplicationConnection]: Initializing PgOutput logical decoder publication
2023-02-13 13:54:57 2023-02-13 12:54:57,091 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [i.d.c.c.BaseSourceTask]: Stopping down connector
2023-02-13 13:54:57 2023-02-13 12:54:57,342 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.c.PostgresConnection]: Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{0/A23A73B8}, catalogXmin=1060498]
2023-02-13 13:54:57 2023-02-13 12:54:57,345 [ INFO] [pool-3-thread-1] [i.d.j.JdbcConnection]: Connection gracefully closed
2023-02-13 13:54:57 2023-02-13 12:54:57,449 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.Threads]: Requested thread factory for connector PostgresConnector, id = 5664d41e-18a6-437f-93d4-d10eb54d2e6c named = keep-alive
2023-02-13 13:54:57 2023-02-13 12:54:57,450 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.Threads]: Creating thread debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-keep-alive
2023-02-13 13:54:57 2023-02-13 12:54:57,681 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresSchema]: REPLICA IDENTITY for 'public.company_profile' is 'DEFAULT'; UPDATE and DELETE events will contain previous values only for PK columns
2023-02-13 13:54:57 2023-02-13 12:54:57,713 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresStreamingChangeEventSource]: Searching for WAL resume position
2023-02-13 13:54:57 2023-02-13 12:54:57,714 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresStreamingChangeEventSource]: WAL resume position 'null' discovered
2023-02-13 13:54:57 2023-02-13 12:54:57,717 [ INFO] [pool-4-thread-1] [i.d.j.JdbcConnection]: Connection gracefully closed
2023-02-13 13:54:57 2023-02-13 12:54:57,720 [ INFO] [pool-5-thread-1] [i.d.j.JdbcConnection]: Connection gracefully closed
2023-02-13 13:54:57 2023-02-13 12:54:57,889 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.c.PostgresReplicationConnection]: Initializing PgOutput logical decoder publication
2023-02-13 13:54:58 2023-02-13 12:54:58,026 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.Threads]: Requested thread factory for connector PostgresConnector, id = 5664d41e-18a6-437f-93d4-d10eb54d2e6c named = keep-alive
2023-02-13 13:54:58 2023-02-13 12:54:58,026 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.u.Threads]: Creating thread debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-keep-alive
2023-02-13 13:54:58 2023-02-13 12:54:58,026 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.c.p.PostgresStreamingChangeEventSource]: Processing messages
2023-02-13 13:54:58 2023-02-13 12:54:58,144 [ INFO] [pool-6-thread-1] [i.d.j.JdbcConnection]: Connection gracefully closed
2023-02-13 13:54:58 2023-02-13 12:54:58,146 [ INFO] [pool-7-thread-1] [i.d.j.JdbcConnection]: Connection gracefully closed
2023-02-13 13:54:58 2023-02-13 12:54:58,146 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Finished streaming
2023-02-13 13:54:58 2023-02-13 12:54:58,147 [ INFO] [debezium-postgresconnector-5664d41e-18a6-437f-93d4-d10eb54d2e6c-change-event-source-coordinator] [i.d.p.ChangeEventSourceCoordinator]: Connected metrics set to 'false'
2023-02-13 13:54:58 2023-02-13 12:54:58,148 [ INFO] [hz.vibrant_curran.jet.blocking.thread-0] [c.h.j.i.e.TaskletExecutionService]: [192.168.0.2]:5701 [hello-world] [5.2.1] Exception in ProcessorTasklet{postgres-monitor-company_profile/source#0}
2023-02-13 13:54:58 com.hazelcast.jet.JetException: Failed to connect to database
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.reconnect(CdcSourceP.java:216) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.complete(CdcSourceP.java:196) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:541) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:421) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:315) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
2023-02-13 13:54:58     at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
2023-02-13 13:54:58     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
2023-02-13 13:54:58     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
2023-02-13 13:54:58     at java.lang.Thread.run(Thread.java:829) ~[?:?]
2023-02-13 13:54:58 Caused by: java.lang.NullPointerException: keyJson
2023-02-13 13:54:58     at java.util.Objects.requireNonNull(Objects.java:246) ~[?:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.ChangeRecordImpl.<init>(ChangeRecordImpl.java:61) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.ChangeRecordCdcSourceP.map(ChangeRecordCdcSourceP.java:102) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.ChangeRecordCdcSourceP.map(ChangeRecordCdcSourceP.java:34) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.lambda$complete$0(CdcSourceP.java:187) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.util.FlatMappingTraverser.nextTraverser(FlatMappingTraverser.java:63) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:55) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.complete(CdcSourceP.java:191) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     ... 9 more
2023-02-13 13:54:58 2023-02-13 12:54:58,163 [ERROR] [ForkJoinPool.commonPool-worker-19] [c.h.j.i.o.StartExecutionOperation]: [192.168.0.2]:5701 [hello-world] [5.2.1] Exception in ProcessorTasklet{postgres-monitor-company_profile/source#0}: com.hazelcast.jet.JetException: Failed to connect to database
2023-02-13 13:54:58 com.hazelcast.jet.JetException: Exception in ProcessorTasklet{postgres-monitor-company_profile/source#0}: com.hazelcast.jet.JetException: Failed to connect to database
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.TaskletExecutionService.handleTaskletExecutionError(TaskletExecutionService.java:286) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.TaskletExecutionService.access$600(TaskletExecutionService.java:80) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:325) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
2023-02-13 13:54:58     at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
2023-02-13 13:54:58     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
2023-02-13 13:54:58     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
2023-02-13 13:54:58     at java.lang.Thread.run(Thread.java:829) ~[?:?]
2023-02-13 13:54:58 Caused by: com.hazelcast.jet.JetException: Failed to connect to database
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.reconnect(CdcSourceP.java:216) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.complete(CdcSourceP.java:196) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:541) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:421) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:315) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     ... 5 more
2023-02-13 13:54:58 Caused by: java.lang.NullPointerException: keyJson
2023-02-13 13:54:58     at java.util.Objects.requireNonNull(Objects.java:246) ~[?:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.ChangeRecordImpl.<init>(ChangeRecordImpl.java:61) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.ChangeRecordCdcSourceP.map(ChangeRecordCdcSourceP.java:102) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.ChangeRecordCdcSourceP.map(ChangeRecordCdcSourceP.java:34) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.lambda$complete$0(CdcSourceP.java:187) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.util.FlatMappingTraverser.nextTraverser(FlatMappingTraverser.java:63) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.util.FlatMappingTraverser.next(FlatMappingTraverser.java:55) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:405) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.core.AbstractProcessor.emitFromTraverser(AbstractProcessor.java:421) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.cdc.impl.CdcSourceP.complete(CdcSourceP.java:191) ~[hazelcast-jet-cdc-debezium-5.2.1.jar:?]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.complete(ProcessorTasklet.java:541) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:421) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:291) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:315) ~[hazelcast-5.2.1.jar:5.2.1]
2023-02-13 13:54:58     ... 5 more
```

--- a/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/ChangeRecord.java
+++ b/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/ChangeRecord.java
@@ -128,7 +128,7 @@ public interface ChangeRecord {
     /**
      * Returns the key part of the CDC event. It identifies the affected record.
      */
-    @Nonnull
+    @Nullable
     RecordPart key();
 
     /**

--- a/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordImpl.java
+++ b/extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordImpl.java
@@ -47,7 +47,7 @@ public class ChangeRecordImpl implements ChangeRecord {
             long sequenceSource,
             long sequenceValue,
             Operation operation,
-            @Nonnull String keyJson,
+            @Nullable String keyJson,
             Supplier<String> oldValueJsonSupplier,
             Supplier<String> newValueJsonSupplier,
             String table,
@@ -58,7 +58,7 @@ public class ChangeRecordImpl implements ChangeRecord {
         this.sequenceSource = sequenceSource;
         this.sequenceValue = sequenceValue;
         this.operation = operation;
-        this.keyJson = requireNonNull(keyJson, "keyJson");
+        this.keyJson = keyJson;
         this.oldValue = oldValueJsonSupplier == null ? null : new RecordPartImpl(oldValueJsonSupplier);
         this.newValue = newValueJsonSupplier == null ? null : new RecordPartImpl(newValueJsonSupplier);
         this.table = table;
@@ -71,7 +71,7 @@ public class ChangeRecordImpl implements ChangeRecord {
             long sequenceSource,
             long sequenceValue,
             Operation operation,
-            @Nonnull String keyJson,
+            String keyJson,
             String oldValueJson,
             String newValueJson,
             String table,
@@ -82,7 +82,7 @@ public class ChangeRecordImpl implements ChangeRecord {
         this.sequenceSource = sequenceSource;
         this.sequenceValue = sequenceValue;
         this.operation = operation;
-        this.keyJson = requireNonNull(keyJson, "keyJson");
+        this.keyJson = keyJson;
         this.oldValue = oldValueJson == null ? null : new RecordPartImpl(oldValueJson);
         this.newValue = newValueJson == null ? null : new RecordPartImpl(newValueJson);
         this.table = table;
@@ -120,9 +120,12 @@ public class ChangeRecordImpl implements ChangeRecord {
     }
 
     @Override
-    @Nonnull
+    @Nullable
     public RecordPart key() {
         if (key == null) {
+            if (keyJson == null) {
+                return null;
+            }
             key = new RecordPartImpl(() -> keyJson);
         }
         return key;

--- a/extensions/cdc-debezium/src/test/java/com/hazelcast/jet/cdc/DebeziumCdcIntegrationTest.java
+++ b/extensions/cdc-debezium/src/test/java/com/hazelcast/jet/cdc/DebeziumCdcIntegrationTest.java
@@ -32,6 +32,7 @@ import com.mongodb.client.MongoClient;
 import com.mongodb.client.MongoClients;
 import io.debezium.connector.mongodb.MongoDbConnector;
 import io.debezium.connector.mysql.MySqlConnector;
+import io.debezium.connector.postgresql.PostgresConnector;
 import org.bson.Document;
 import org.junit.Assume;
 import org.junit.Test;
@@ -136,7 +137,7 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
         return DebeziumCdcSources.debezium("mysql",
                         MySqlConnector.class)
                         .setProperty("include.schema.changes", "true")
-                        .setProperty("database.hostname", container.getContainerIpAddress())
+                        .setProperty("database.hostname", container.getHost())
                         .setProperty("database.port", Integer.toString(container.getMappedPort(MYSQL_PORT)))
                         .setProperty("database.user", "debezium")
                         .setProperty("database.password", "dbz")
@@ -199,7 +200,7 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
             StreamSource<Entry<String, String>> source = DebeziumCdcSources.debeziumJson("mysql",
                     MySqlConnector.class)
                     .setProperty("include.schema.changes", "false")
-                    .setProperty("database.hostname", container.getContainerIpAddress())
+                    .setProperty("database.hostname", container.getHost())
                     .setProperty("database.port", Integer.toString(container.getMappedPort(MYSQL_PORT)))
                     .setProperty("database.user", "debezium")
                     .setProperty("database.password", "dbz")
@@ -256,7 +257,7 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
             StreamSource<ChangeRecord> source = DebeziumCdcSources.debezium("postgres",
                     "io.debezium.connector.postgresql.PostgresConnector")
                     .setProperty("database.server.name", "dbserver1")
-                    .setProperty("database.hostname", container.getContainerIpAddress())
+                    .setProperty("database.hostname", container.getHost())
                     .setProperty("database.port", Integer.toString(container.getMappedPort(POSTGRESQL_PORT)))
                     .setProperty("database.user", "postgres")
                     .setProperty("database.password", "postgres")
@@ -352,7 +353,7 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
             StreamSource<Entry<String, String>> source = DebeziumCdcSources.debeziumJson("postgres",
                     "io.debezium.connector.postgresql.PostgresConnector")
                     .setProperty("database.server.name", "dbserver1")
-                    .setProperty("database.hostname", container.getContainerIpAddress())
+                    .setProperty("database.hostname", container.getHost())
                     .setProperty("database.port", Integer.toString(container.getMappedPort(POSTGRESQL_PORT)))
                     .setProperty("database.user", "postgres")
                     .setProperty("database.password", "postgres")
@@ -435,6 +436,42 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
             hz.getJet().newJob(pipeline);
 
             assertTrueEventually(() -> assertThat(hz.getList("notFailWhenOldValueNotPresent")).isNotEmpty());
+        }
+    }
+    @Test
+    public void noFailWhenNoPrimaryKey() throws Exception {
+        try (PostgreSQLContainer<?> container = postgresContainer()) {
+            container.start();
+            //when
+            try (Connection connection = getPostgreSqlConnection(container.getJdbcUrl(), container.getUsername(),
+                    container.getPassword())) {
+                connection.setSchema("inventory");
+                Statement statement = connection.createStatement();
+                statement.addBatch("CREATE TABLE NO_PK (SOME_INT INT);");
+                statement.addBatch("INSERT INTO NO_PK VALUES (1)");
+                statement.executeBatch();
+            }
+
+            Pipeline pipeline = Pipeline.create();
+
+            StreamSource<ChangeRecord> source = DebeziumCdcSources
+                    .debezium("postgres", PostgresConnector.class)
+                    .setProperty("database.server.name", "dbserver1")
+                    .setProperty("database.hostname", container.getHost())
+                    .setProperty("database.port", Integer.toString(container.getMappedPort(POSTGRESQL_PORT)))
+                    .setProperty("database.user", "postgres")
+                    .setProperty("database.password", "postgres")
+                    .setProperty("database.dbname", "postgres")
+                    .setProperty("table.whitelist", "inventory.no_pk")
+                    .build();
+            pipeline.readFrom(source)
+                    .withNativeTimestamps(1)
+                    .writeTo(Sinks.list("no_pk"));
+
+            HazelcastInstance hz = createHazelcastInstances(1)[0];
+            hz.getJet().newJob(pipeline);
+
+            assertTrueEventually(() -> assertThat(hz.getList("no_pk")).isNotEmpty());
         }
     }
 

--- a/extensions/cdc-debezium/src/test/java/com/hazelcast/jet/cdc/DebeziumCdcIntegrationTest.java
+++ b/extensions/cdc-debezium/src/test/java/com/hazelcast/jet/cdc/DebeziumCdcIntegrationTest.java
@@ -113,8 +113,9 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
 
             //when
             try (Connection connection = getMySqlConnection(container.withDatabaseName("inventory").getJdbcUrl(),
-                    container.getUsername(), container.getPassword())) {
-                Statement statement = connection.createStatement();
+                    container.getUsername(), container.getPassword());
+                 Statement statement = connection.createStatement()
+            ) {
                 statement.addBatch("UPDATE customers SET first_name='Anne Marie' WHERE id=1004");
                 statement.addBatch("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
                 statement.addBatch("DELETE FROM customers WHERE id=1005");
@@ -291,11 +292,12 @@ public class DebeziumCdcIntegrationTest extends AbstractCdcIntegrationTest {
             try (Connection connection = getPostgreSqlConnection(container.getJdbcUrl(), container.getUsername(),
                     container.getPassword())) {
                 connection.setSchema("inventory");
-                Statement statement = connection.createStatement();
-                statement.addBatch("UPDATE customers SET first_name='Anne Marie' WHERE id=1004");
-                statement.addBatch("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
-                statement.addBatch("DELETE FROM customers WHERE id=1005");
-                statement.executeBatch();
+                try (Statement statement = connection.createStatement()) {
+                    statement.addBatch("UPDATE customers SET first_name='Anne Marie' WHERE id=1004");
+                    statement.addBatch("INSERT INTO customers VALUES (1005, 'Jason', 'Bourne', 'jason@bourne.org')");
+                    statement.addBatch("DELETE FROM customers WHERE id=1005");
+                    statement.executeBatch();
+                }
             }
 
             //then

